alibi_detect/utils/warnings.py

```diff
@@ -0,0 +1,24 @@
+import functools
```

Comment:
Would be good to add a module docstring as well as docstrings to both functions.